### ISSUE TITLE
Apply update and restart every server instance?

### ISSUE THREAD

**Issue by fs86** [GameServerManagers/LinuxGSM#1943]
I have a DST server (master and caves) which are connected to each other. Both instances run under the same user. When an update is applied with "./dstserver-1 update", server-1 will be restarted when the update process has finished. Unfortenately the second instance (which is my Caves server) doesn't get restarted and runs on an old server version. Is there already a possibility to force each server instance to restart after an updated has been applied?

**Comment by UltimateByte** [comment]
This would likely require a massive overhaul of LinuxGSM and the way it manages instances to be accomplished directly. But the workaround is easy:
Use force-update on the first server, and stop on the second, then about one or two minutes later (you know the time it takes to update better than me), start the second server. You can do all that using cronjobs.
https://github.com/GameServerManagers/LinuxGSM/wiki/Cronjobs

**Comment by fs86** [comment]
Yes, that should work, but this would result in a server restart even when there is no new update. And I want to prevent a server disconnect when it is not necessary.

**Comment by UltimateByte** [comment]
In fact, what you're proposing would benefit everyone with multiple instances under the same install.
We could add a parameter to stop and restart all instances upon update.
The question is: How will it detect other instances.
Maybe we could make a separate script for that, like "linuxgsm-instancesmanage.sh" or something like that.

**Comment by fs86** [comment]
What about adding a configuration file with a grouped list of instances? If "./gameserver-1 update" is executed, the script can look for other gameserver instances within the same group as "gameserver-1" and perform a restart.

**Comment by UltimateByte** [comment]
That's a design choice to make.

I suggest a dedicated script like `linuxgsm-instances.sh` which would be able to detect instances, then manage them from one script. There would be a master instance (for updates), then slaves. It would loop through bash scripts in rootdir and ask to validate a list. You could input commands like `linuxgsm-instances.sh start instancename` or `linuxgsm-instances.sh restart all`, and it would be smart enough so that if you run `update` or `validate` it knows it needs to run it only on the main instance and shutdown the others when an update is found or upon validating.

What do you think?

@dgibbs64 Would you do it, and if yes, would you validate my idea?

**Comment by dgibbs64** [comment]
@UltimateByte I think its something we have talked about before using `./linuxgsm.sh` to manage it. Sounds like a great idea and it should be relatively easy for the instances script to identify instances within its own dir. Would be a great way to practice using milestones as well :)

**Comment by UltimateByte** [comment]
Oh yeah, managing directly within linuxgsm.sh would be the greatest for sure, forgot about that idea.
Let's talk about milestones on Discord! :D
